### PR TITLE
Un revert infert rt nola patch

### PR DESCRIPTION
**Pull request by TomGoBravo**
## This PR

This unreverts https://github.com/covid-projections/covid-data-model/pull/735, in other words it is a second attempt at https://github.com/covid-projections/covid-data-model/pull/731 this time with fix for `KeyError <CommonFields.ICU_BEDS: 'icu_beds'>` reported at https://sentry.io/organizations/covidactnow/issues/1957780407/?project=5203044&referrer=slack

Changes `OneRegionTimeseriesDataset.data` to always contain location_id.

Changes `MultiRegionTimeseriesDataset.data` to *not* have FIPS, forcing callers that want it to use `data_with_fips`.

### Please Check
- [ ] Are the [JSON schemas](https://github.com/covid-projections/covid-data-model/tree/master/api/schemas) updated?
- [ ] Are the [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.V1.md) updated?
- [ ] Are tests passing?

### CODE REVIEW

File: libs/datasets/timeseries.py [hunk 1]
```diff
@@ -113,14 +107,20 @@ def yield_records(self) -> Iterable[dict]:
 
     def get_subset(self, after=None, columns=tuple()):
         rows_key = dataset_utils.make_rows_key(self.data, after=after,)
-        columns_key = list(columns) if columns else slice(None, None, None)
+        if columns:
+            assert CommonFields.LOCATION_ID in columns
```
Comment (ghop02) [inline libs/datasets/timeseries.py:111]: I think if we have these asserts in the input data itself it makes it easier to reason about it elsewhere.  Why did you move it from the post init to here?

**Review by ghop02 — APPROVED** [review]
I think this is a step in the right direction and largely agrees with some of the work that I am doing to get MSAs into the API.

File: libs/datasets/timeseries.py (hunk 1 above)
Comment (TomGoBravo) [reply libs/datasets/timeseries.py:111]: I bumped into a bug where LOCATION_ID wasn't in columns but you are correct that the check makes more sense in `__post_init__` so moved them there.
